### PR TITLE
planner: fix error result when update bigint unsigned column to -1

### LINKED ISSUE #8623: error result when update bigint unsigned column to -1
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?

```sql
create table t (a bigint unsigned);
insert into t value (1);
update t set a = -1;
```

2. What did you expect to see?
```sql
mysql> insert into t value (1);
Query OK, 1 row affected (0.07 sec)

mysql> update t set a = -1;
ERROR 1264 (22003): Out of range value for column 'a' at row 1
```

3. What did you see instead?
```sql
mysql> insert into t value (1);
Query OK, 1 row affected (0.01 sec)

mysql> update t set a = -1;
Query OK, 1 row affected (0.00 sec)

mysql> select * from t;
+----------------------+
| a                    |
+----------------------+
| 18446744073709551615 |
+----------------------+
1 row in set (0.01 sec)
```

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

```sql
mysql> select tidb_version()\G
*************************** 1. row ***************************
tidb_version(): Release Version: v2.1.0-rc.3-294-g6e2d6c7aa
Git Commit Hash: 6e2d6c7aa7eba3ac4f3a5201f1a36bf534fa6298
Git Branch: master
UTC Build Time: 2018-12-08 05:24:33
GoVersion: go version go1.11.2 linux/amd64
Race Enabled: false
TiKV Min Version: 2.1.0-alpha.1-ff3dd160846b7d1aed9079c389fc188f7f5ea13e
Check Table Before Drop: false
```

## SIG slack channel
[#sig-exec](https://tidbcommunity.slack.com/archives/CMRD79DRR)

## Score
- 300

## Mentor
 * @qw4990

### PR DESCRIPTION
**Pull request by exialin**
### What problem does this PR solve? 
Fix #8623 .

### What is changed and how it works?
The following cast seems redundant:
https://github.com/pingcap/tidb/blob/640aa6511316696e499450e9c1e927ce11ecab75/planner/core/logical_plan_builder.go#L2475

because the value will be evaluated during execution:
https://github.com/pingcap/tidb/blob/640aa6511316696e499450e9c1e927ce11ecab75/executor/update.go#L233

### Check List 

Tests 

 - Unit test

Code changes

 - Minor change

Side effects

 - None

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: executor/write_test.go [hunk 1]
```diff
@@ -1330,7 +1330,26 @@ func (s *testSuite) TestUpdate(c *C) {
 	tk.MustExec("update t set a = ''")
 	tk.MustQuery("select * from t").Check(testkit.Rows("0000-00-00 00:00:00 1999-12-13 00:00:00"))
 	tk.MustExec("update t set b = ''")
-	tk.MustQuery("select * from t").Check(testkit.Rows("0000-00-00 00:00:00 <nil>"))
+	tk.MustQuery("select * from t").Check(testkit.Rows("0000-00-00 00:00:00 0000-00-00 00:00:00"))
```
Comment (xiekeyi98) [inline executor/write_test.go]: I was wondering if I could know why we need change this line?

Comment (exialin) [reply executor/write_test.go]: It is consistent with MySQL now. The following is the result of MySQL:
```sql
mysql> create table t(a datetime not null, b datetime);
Query OK, 0 rows affected (0.33 sec)

mysql> insert into t value('1999-12-12', '1999-12-13') ;
Query OK, 1 row affected (0.13 sec)

mysql> set sql_mode='';
Query OK, 0 rows affected, 1 warning (0.00 sec)

mysql> update t set b = '';
Query OK, 1 row affected, 1 warning (0.12 sec)
Rows matched: 1  Changed: 1  Warnings: 1

mysql> select * from t;
+---------------------+---------------------+
| a                   | b                   |
+---------------------+---------------------+
| 1999-12-12 00:00:00 | 0000-00-00 00:00:00 |
+---------------------+---------------------+
1 row in set (0.00 sec)
```

**Comment by zz-jason** [comment]
@exialin Sorry for the late review. Could you update the master branch and resolve the conflicts?

**Comment by zz-jason** [comment]
@exialin CI failed, could you please take your spare time to fix it in order to make this PR reviewable?

**Comment by zz-jason** [comment]
Close due to outdated, fell free to reopen it if you are going to work on this PR in the future.

**Pull request closed without merging**